pkg/scheduler/scheduler.go

```diff
@@ -221,14 +227,21 @@ func (s *Scheduler) Run(ctx context.Context) {
 	// Starting the scheduling queue.
 	s.queue.Run()
 
-	// Run scheduleOnce forever.
-	//
-	// The loop starts in a dedicated goroutine; it exits when the context is canceled.
-	go wait.UntilWithContext(ctx, s.scheduleOnce, 0)
+	wg := &sync.WaitGroup{}
+	wg.Add(s.workerNumber)
+	for i := 0; i < s.workerNumber; i++ {
+		go func() {
+			defer wg.Done()
+			defer utilruntime.HandleCrash()
+			// Run scheduleOnce forever.
+			s.scheduleOnce(ctx)
```

Comment:
Hi Ryan! I think the test is failing because each worker only run once and then exits